pkg/appgw/requestroutingrules.go

```diff
@@ -365,15 +365,35 @@ func (c *appGwConfigBuilder) mergePathMap(existingPathMap *n.ApplicationGatewayU
 		existingPathMap.DefaultBackendAddressPool = nil
 		existingPathMap.DefaultBackendHTTPSettings = nil
 	}
-	if pathMapToMerge.PathRules == nil || len(*pathMapToMerge.PathRules) == 0 {
```

Comment:
is there a reason this check was removed ?

Reply:
We added a similar condition in a new if block, but yeah makes sense to preserve the original condition.